crates/rpc/src/jsonrpc/request.rs

```diff
@@ -48,7 +48,7 @@ impl<'a> RawParams<'a> {
     pub fn deserialize<T: Deserialize<'a>>(self) -> Result<T, RpcError> {
         let s = self.0.map(|x| x.get()).unwrap_or_default();
 
-        serde_json::from_str::<T>(s).map_err(|_| RpcError::InvalidParams)
+        serde_json::from_str::<T>(s).map_err(|e| RpcError::InvalidParams(e.to_string()))
```

Comment:
It feels wrong to return an `RpcError` when the only possible error is a single variant.
This leads for example to useless match branches in the Websocket logic.
Maybe we should introduce an error struct for this case?

Anyway, not a blocking issue. Feel free to leave that as a TODO or to completely disregard it.

Reply:
Yeah its a fair point. I think I originally added it so one doesn't have to repeat the mapping everywhere. But at this point I think one is mapping and de-mapping so could be worth removing it again.